config/configtls/configtls.go

```diff
@@ -327,24 +330,25 @@ func (c TLSClientSetting) LoadTLSConfig() (*tls.Config, error) {
 }
 
 // LoadTLSConfig loads the TLS configuration.
-func (c TLSServerSetting) LoadTLSConfig() (*tls.Config, error) {
+func (c *TLSServerSetting) LoadTLSConfig() (*tls.Config, error) {
 	tlsCfg, err := c.loadTLSConfig()
 	if err != nil {
 		return nil, fmt.Errorf("failed to load TLS config: %w", err)
 	}
 	if c.ClientCAFile != "" {
-		reloader, err := newClientCAsReloader(c.ClientCAFile, &c)
+		var err error
+		c.reloader, err = newClientCAsReloader(c.ClientCAFile, c)
 		if err != nil {
 			return nil, err
 		}
 		if c.ReloadClientCAFile {
-			err = reloader.startWatching()
+			err = c.reloader.startWatching()
```

Comment:
Strange logic that we create a reloader even when not starting, maybe split the logic? Separate PR would be preferred for that.

Reply:
To make sure I understand, you're saying it's possible to get here even when the collector is not starting, so we should fix this logic to make sure the reloader only starts watching during start, am I following your thought?